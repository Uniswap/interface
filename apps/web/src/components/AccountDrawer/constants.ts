import { atom } from 'jotai'

export enum MenuState {
  DEFAULT = 'default',
  SETTINGS = 'settings',
  LANGUAGE_SETTINGS = 'language_settings',
  LOCAL_CURRENCY_SETTINGS = 'local_currency_settings',
  LIMITS = 'limits',
  POOLS = 'pools',
  PASSKEYS = 'passkeys',
  OTHER_WALLETS = 'other_wallets',
}

export const miniPortfolioMenuStateAtom = atom(MenuState.DEFAULT)
