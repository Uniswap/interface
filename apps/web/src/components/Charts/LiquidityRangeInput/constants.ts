import { isMobileWeb } from 'utilities/src/platform'

const RIGHT_AXIS_WIDTH = 64
const CHART_CONTAINER_WIDTH = 452 + RIGHT_AXIS_WIDTH
const LIQUIDITY_CHART_WIDTH = 68
const INTER_CHART_PADDING = 12
const CHART_HEIGHT = 164
const BOTTOM_AXIS_HEIGHT = 28
const loadedPriceChartWidth = CHART_CONTAINER_WIDTH - LIQUIDITY_CHART_WIDTH - INTER_CHART_PADDING - RIGHT_AXIS_WIDTH

const desktopSizes = {
  rightAxisWidth: RIGHT_AXIS_WIDTH,
  chartContainerWidth: CHART_CONTAINER_WIDTH,
  liquidityChartWidth: LIQUIDITY_CHART_WIDTH,
  interChartPadding: INTER_CHART_PADDING,
  chartHeight: CHART_HEIGHT,
  bottomAxisHeight: BOTTOM_AXIS_HEIGHT,
  loadedPriceChartWidth,
}

const mobileSizes = {
  rightAxisWidth: 0,
  chartContainerWidth: 290,
  liquidityChartWidth: 48,
  interChartPadding: 0,
  chartHeight: CHART_HEIGHT,
  bottomAxisHeight: BOTTOM_AXIS_HEIGHT,
  loadedPriceChartWidth: 290,
}

export function useRangeInputSizes(parentWidth?: number) {
  return isMobileWeb
    ? {
        ...mobileSizes,
        chartContainerWidth: parentWidth ?? mobileSizes.chartContainerWidth,
        loadedPriceChartWidth: parentWidth ?? mobileSizes.loadedPriceChartWidth,
      }
    : desktopSizes
}
