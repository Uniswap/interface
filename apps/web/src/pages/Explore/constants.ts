export enum ExploreTab {
  Tokens = 'tokens',
  Pools = 'pools',
  Transactions = 'transactions',
}
