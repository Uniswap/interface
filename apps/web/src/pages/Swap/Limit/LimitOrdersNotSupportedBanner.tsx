import AlertTriangleFilled from 'components/Icons/AlertTriangleFilled'
import { useTranslation } from 'react-i18next'
import { Flex, Text, TouchableArea } from 'ui/src'

interface LimitOrdersNotSupportedBannerProps {
  onMoreDetails: () => void
}

function LimitOrdersNotSupportedBanner({ onMoreDetails }: LimitOrdersNotSupportedBannerProps) {
  const { t } = useTranslation()

  return (
    <Flex row backgroundColor="$surface2" borderRadius="$rounded12" p="$padding12" gap="$spacing12">
      <AlertTriangleFilled width={20} height={20} color="$neutral1" />
      <Flex row $sm={{ flexDirection: 'column' }}>
        <Text variant="body3" color="$neutral1">
          {`${t('smartWallets.delegationMismatchModal.limitOrders.unsupported')} `}
        </Text>
        <TouchableArea onPress={onMoreDetails}>
          <Text variant="body3" color="$accent1">
            {t('common.moreDetails')}
          </Text>
        </TouchableArea>
      </Flex>
    </Flex>
  )
}

export default LimitOrdersNotSupportedBanner
