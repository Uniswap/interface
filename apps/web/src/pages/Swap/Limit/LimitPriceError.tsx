import { Currency } from '@uniswap/sdk-core'
import { LimitPriceErrorType } from 'components/CurrencyInputPanel/LimitPriceInputPanel/useCurrentPriceAdjustment'
import Column from 'components/deprecated/Column'
import Row from 'components/deprecated/Row'
import styled, { useTheme } from 'lib/styled-components'
import { ReactNode } from 'react'
import { AlertTriangle } from 'react-feather'
import { ThemedText } from 'theme/components'
import { FadePresence, FadePresenceAnimationType } from 'theme/components/FadePresence'
import { Trans } from 'uniswap/src/i18n'

const Container = styled(Row)`
  padding: 12px;
  border: 1px solid ${({ theme }) => theme.surface3};
  border-radius: 16px;
  margin-top: 4px;
`

const LogoContainer = styled.div`
  height: 40px;
  width: 40px;
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 12px;
  background-color: ${({ theme }) => theme.critical2};
  flex-shrink: 0;
`

interface LimitPriceErrorProps {
  priceError: LimitPriceErrorType
  inputCurrency: Currency
  outputCurrency: Currency
  priceInverted: boolean
  priceAdjustmentPercentage?: number
}

function getTitle({ inputCurrency, outputCurrency, priceInverted, priceError }: LimitPriceErrorProps): ReactNode {
  if (priceError === LimitPriceErrorType.CALCULATION_ERROR) {
    return <Trans i18nKey="limitPrice.marketPriceNotAvailable.error.title" />
  } else if (priceInverted) {
    return (
      <Trans i18nKey="limitPrice.buyingAboveMarketPrice.error.title" values={{ tokenSymbol: outputCurrency.symbol }} />
    )
  } else {
    return (
      <Trans i18nKey="limitPrice.sellingBelowMarketPrice.error.title" values={{ tokenSymbol: inputCurrency.symbol }} />
    )
  }
}

function getDescription({ priceInverted, priceAdjustmentPercentage, priceError }: LimitPriceErrorProps): ReactNode {
  if (priceError === LimitPriceErrorType.CALCULATION_ERROR) {
    return (
      <Trans i18nKey="limitPrice.marketPriceNotAvailable.error.description">
        We are unable to calculate the current market price. To avoid submitting an order below market price, please
        check your network connection and try again.
      </Trans>
    )
  } else if (priceInverted && !!priceAdjustmentPercentage) {
    return (
      <Trans
        i18nKey="limitPrice.buyingAboveMarketPrice.error.description"
        values={{ percentage: Math.abs(priceAdjustmentPercentage) }}
      />
    )
  } else if (priceAdjustmentPercentage) {
    return (
      <Trans
        i18nKey="limitPrice.sellingBelowMarketPrice.error.description"
        values={{ percentage: Math.abs(priceAdjustmentPercentage) }}
      />
    )
  }
  return null
}

export function LimitPriceError(props: LimitPriceErrorProps) {
  const theme = useTheme()
  return (
    <FadePresence
      $transitionDuration={theme.transition.duration.fast}
      $delay={theme.transition.duration.fast}
      animationType={FadePresenceAnimationType.FadeAndTranslate}
    >
      <Container gap="md">
        <LogoContainer>
          <AlertTriangle strokeWidth={1} color={theme.critical} size="20px" />
        </LogoContainer>
        <Column>
          <ThemedText.SubHeader>{getTitle(props)}</ThemedText.SubHeader>
          <ThemedText.BodySmall color="neutral2">{getDescription(props)}</ThemedText.BodySmall>
        </Column>
      </Container>
    </FadePresence>
  )
}
