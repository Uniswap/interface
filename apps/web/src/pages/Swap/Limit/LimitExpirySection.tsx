import Row from 'components/deprecated/Row'
import { TFunction } from 'i18next'
import styled from 'lib/styled-components'
import { Trans, useTranslation } from 'react-i18next'
import { useLimitContext } from 'state/limit/LimitContext'
import { ClickableStyle, ThemedText } from 'theme/components'
import { InterfaceEventNameLocal } from 'uniswap/src/features/telemetry/constants'
import { sendAnalyticsEvent } from 'uniswap/src/features/telemetry/send'
import { LimitsExpiry } from 'uniswap/src/types/limits'

const ExpirySection = styled(Row)`
  width: 100%;
  padding: 12px 16px;
  justify-content: space-between;
`

const LimitExpiryButton = styled.button<{ $selected: boolean }>`
  display: flex;
  padding: 4px 8px;
  justify-content: flex-end;
  align-items: center;
  gap: 4px;
  border: 1px solid ${({ theme }) => theme.surface3};
  height: 28px;
  border-radius: 999px;
  background-color: ${({ theme, $selected }) => ($selected ? theme.surface3 : 'unset')};
  color: ${({ theme, $selected }) => ($selected ? theme.neutral1 : theme.neutral2)};
  ${ClickableStyle}
`

const EXPIRY_OPTIONS = [LimitsExpiry.Day, LimitsExpiry.Week, LimitsExpiry.Month, LimitsExpiry.Year]

// eslint-disable-next-line consistent-return
function getExpiryLabelText(t: TFunction, expiry: LimitsExpiry): string {
  switch (expiry) {
    case LimitsExpiry.Day:
      return t('common.oneDay')
    case LimitsExpiry.Week:
      return t('common.oneWeek')
    case LimitsExpiry.Month:
      return t('common.oneMonth')
    case LimitsExpiry.Year:
      return t('common.oneYear')
  }
}

export function LimitExpirySection() {
  const { t } = useTranslation()
  const { limitState, setLimitState } = useLimitContext()

  return (
    <ExpirySection>
      <ThemedText.SubHeaderSmall>
        <Trans i18nKey="common.expiry" />
      </ThemedText.SubHeaderSmall>
      <Row justify="flex-end" gap="xs">
        {EXPIRY_OPTIONS.map((expiry) => (
          <LimitExpiryButton
            key={expiry}
            $selected={expiry === limitState.expiry}
            onClick={() => {
              if (expiry === limitState.expiry) {
                return
              }
              sendAnalyticsEvent(InterfaceEventNameLocal.LimitExpirySelected, {
                value: expiry,
              })
              setLimitState((prev) => ({
                ...prev,
                expiry,
              }))
            }}
          >
            <ThemedText.LabelSmall color="inherit" fontWeight={535}>
              {getExpiryLabelText(t, expiry)}
            </ThemedText.LabelSmall>
          </LimitExpiryButton>
        ))}
      </Row>
    </ExpirySection>
  )
}
