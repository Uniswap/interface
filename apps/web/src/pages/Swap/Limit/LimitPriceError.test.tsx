import { LimitPriceErrorType } from 'components/CurrencyInputPanel/LimitPriceInputPanel/useCurrentPriceAdjustment'
import { LimitPriceError } from 'pages/Swap/Limit/LimitPriceError'
import { render } from 'test-utils/render'
import { DAI, USDC_MAINNET } from 'uniswap/src/constants/tokens'

describe('LimitPriceError', () => {
  it.each([
    [true, 10],
    [false, 10],
    [true, -10],
    [false, -10],
  ])('renders the limit price error correctly, inverted %p change %p', async (inverted, change) => {
    const { container } = render(
      <LimitPriceError
        inputCurrency={DAI}
        outputCurrency={USDC_MAINNET}
        priceInverted={inverted}
        priceAdjustmentPercentage={change}
        priceError={LimitPriceErrorType.BELOW_MARKET}
      />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders the limit price error correctly when there is a calculation error', async () => {
    const { container } = render(
      <LimitPriceError
        inputCurrency={DAI}
        outputCurrency={USDC_MAINNET}
        priceInverted={false}
        priceAdjustmentPercentage={0}
        priceError={LimitPriceErrorType.CALCULATION_ERROR}
      />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
