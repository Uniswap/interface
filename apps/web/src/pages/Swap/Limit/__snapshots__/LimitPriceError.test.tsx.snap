// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LimitPriceError renders the limit price error correctly when there is a calculation error 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c6 {
  color: #222222;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c7 {
  color: #7D7D7D;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c0 {
  -webkit-transition: display 125ms ease-in-out, -webkit-transform 125ms ease-in-out;
  -webkit-transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  -webkit-animation: gyvpAI 125ms ease-in-out forwards;
  animation: gyvpAI 125ms ease-in-out forwards;
  -webkit-animation-delay: 125ms;
  animation-delay: 125ms;
  opacity: 0;
  -webkit-translate: translateY(10px);
  translate: translateY(10px);
}

.c0.exiting {
  -webkit-animation: dnSqss 125ms ease-in-out;
  animation: dnSqss 125ms ease-in-out;
}

.c3 {
  padding: 12px;
  border: 1px solid #22222212;
  border-radius: 16px;
  margin-top: 4px;
}

.c4 {
  height: 40px;
  width: 40px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 12px;
  background-color: #FFF2F1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<span
  class=""
  style="display: contents;"
>
  <span
    class=" _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="c0"
      >
        <div
          class="c1 c2 c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="20px"
              stroke="#FF5F52"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
              />
              <line
                x1="12"
                x2="12"
                y1="9"
                y2="13"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 css-n8z49y"
            >
              Market price not available
            </div>
            <div
              class="c7 css-142zc9n"
            >
              We are unable to calculate the current market price. To avoid submitting an order below market price, please check your network connection and try again.
            </div>
          </div>
        </div>
      </div>
    </span>
  </span>
</span>
`;

exports[`LimitPriceError renders the limit price error correctly, inverted false change -10 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c6 {
  color: #222222;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c7 {
  color: #7D7D7D;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c0 {
  -webkit-transition: display 125ms ease-in-out, -webkit-transform 125ms ease-in-out;
  -webkit-transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  -webkit-animation: gyvpAI 125ms ease-in-out forwards;
  animation: gyvpAI 125ms ease-in-out forwards;
  -webkit-animation-delay: 125ms;
  animation-delay: 125ms;
  opacity: 0;
  -webkit-translate: translateY(10px);
  translate: translateY(10px);
}

.c0.exiting {
  -webkit-animation: dnSqss 125ms ease-in-out;
  animation: dnSqss 125ms ease-in-out;
}

.c3 {
  padding: 12px;
  border: 1px solid #22222212;
  border-radius: 16px;
  margin-top: 4px;
}

.c4 {
  height: 40px;
  width: 40px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 12px;
  background-color: #FFF2F1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<span
  class=""
  style="display: contents;"
>
  <span
    class=" _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="c0"
      >
        <div
          class="c1 c2 c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="20px"
              stroke="#FF5F52"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
              />
              <line
                x1="12"
                x2="12"
                y1="9"
                y2="13"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 css-n8z49y"
            >
              Selling DAI below market price
            </div>
            <div
              class="c7 css-142zc9n"
            >
              Your limit price is 10% lower than market. Adjust your limit price to proceed.
            </div>
          </div>
        </div>
      </div>
    </span>
  </span>
</span>
`;

exports[`LimitPriceError renders the limit price error correctly, inverted false change 10 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c6 {
  color: #222222;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c7 {
  color: #7D7D7D;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c0 {
  -webkit-transition: display 125ms ease-in-out, -webkit-transform 125ms ease-in-out;
  -webkit-transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  -webkit-animation: gyvpAI 125ms ease-in-out forwards;
  animation: gyvpAI 125ms ease-in-out forwards;
  -webkit-animation-delay: 125ms;
  animation-delay: 125ms;
  opacity: 0;
  -webkit-translate: translateY(10px);
  translate: translateY(10px);
}

.c0.exiting {
  -webkit-animation: dnSqss 125ms ease-in-out;
  animation: dnSqss 125ms ease-in-out;
}

.c3 {
  padding: 12px;
  border: 1px solid #22222212;
  border-radius: 16px;
  margin-top: 4px;
}

.c4 {
  height: 40px;
  width: 40px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 12px;
  background-color: #FFF2F1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<span
  class=""
  style="display: contents;"
>
  <span
    class=" _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="c0"
      >
        <div
          class="c1 c2 c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="20px"
              stroke="#FF5F52"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
              />
              <line
                x1="12"
                x2="12"
                y1="9"
                y2="13"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 css-n8z49y"
            >
              Selling DAI below market price
            </div>
            <div
              class="c7 css-142zc9n"
            >
              Your limit price is 10% lower than market. Adjust your limit price to proceed.
            </div>
          </div>
        </div>
      </div>
    </span>
  </span>
</span>
`;

exports[`LimitPriceError renders the limit price error correctly, inverted true change -10 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c6 {
  color: #222222;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c7 {
  color: #7D7D7D;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c0 {
  -webkit-transition: display 125ms ease-in-out, -webkit-transform 125ms ease-in-out;
  -webkit-transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  -webkit-animation: gyvpAI 125ms ease-in-out forwards;
  animation: gyvpAI 125ms ease-in-out forwards;
  -webkit-animation-delay: 125ms;
  animation-delay: 125ms;
  opacity: 0;
  -webkit-translate: translateY(10px);
  translate: translateY(10px);
}

.c0.exiting {
  -webkit-animation: dnSqss 125ms ease-in-out;
  animation: dnSqss 125ms ease-in-out;
}

.c3 {
  padding: 12px;
  border: 1px solid #22222212;
  border-radius: 16px;
  margin-top: 4px;
}

.c4 {
  height: 40px;
  width: 40px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 12px;
  background-color: #FFF2F1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<span
  class=""
  style="display: contents;"
>
  <span
    class=" _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="c0"
      >
        <div
          class="c1 c2 c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="20px"
              stroke="#FF5F52"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
              />
              <line
                x1="12"
                x2="12"
                y1="9"
                y2="13"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 css-n8z49y"
            >
              Buying USDC above market price
            </div>
            <div
              class="c7 css-142zc9n"
            >
              Your limit price is 10% higher than market. Adjust your limit price to proceed.
            </div>
          </div>
        </div>
      </div>
    </span>
  </span>
</span>
`;

exports[`LimitPriceError renders the limit price error correctly, inverted true change 10 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 12px;
}

.c6 {
  color: #222222;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c7 {
  color: #7D7D7D;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
}

.c0 {
  -webkit-transition: display 125ms ease-in-out, -webkit-transform 125ms ease-in-out;
  -webkit-transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  transition: display 125ms ease-in-out, transform 125ms ease-in-out;
  -webkit-animation: gyvpAI 125ms ease-in-out forwards;
  animation: gyvpAI 125ms ease-in-out forwards;
  -webkit-animation-delay: 125ms;
  animation-delay: 125ms;
  opacity: 0;
  -webkit-translate: translateY(10px);
  translate: translateY(10px);
}

.c0.exiting {
  -webkit-animation: dnSqss 125ms ease-in-out;
  animation: dnSqss 125ms ease-in-out;
}

.c3 {
  padding: 12px;
  border: 1px solid #22222212;
  border-radius: 16px;
  margin-top: 4px;
}

.c4 {
  height: 40px;
  width: 40px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 12px;
  background-color: #FFF2F1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

<span
  class=""
  style="display: contents;"
>
  <span
    class=" _dsp_contents"
  >
    <span
      class=" t_light _dsp_contents is_Theme"
    >
      <div
        class="c0"
      >
        <div
          class="c1 c2 c3"
        >
          <div
            class="c4"
          >
            <svg
              fill="none"
              height="20px"
              stroke="#FF5F52"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
              />
              <line
                x1="12"
                x2="12"
                y1="9"
                y2="13"
              />
              <line
                x1="12"
                x2="12.01"
                y1="17"
                y2="17"
              />
            </svg>
          </div>
          <div
            class="c5"
          >
            <div
              class="c6 css-n8z49y"
            >
              Buying USDC above market price
            </div>
            <div
              class="c7 css-142zc9n"
            >
              Your limit price is 10% higher than market. Adjust your limit price to proceed.
            </div>
          </div>
        </div>
      </div>
    </span>
  </span>
</span>
`;
